starknet_py/net/account/account.py

```diff
@@ -0,0 +1,401 @@
+        if self._client.net not in [TESTNET, MAINNET]:
+            raise ValueError(
+                "Token_address must be specified when using a custom net address"
+            )
+
+        return FEE_CONTRACT_ADDRESS
+
+    async def _get_max_fee(
+        self,
+        transaction: Union[InvokeFunction, Declare, DeployAccount],
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> int:
+        if auto_estimate and max_fee is not None:
+            raise ValueError(
+                "Max_fee and auto_estimate are exclusive and cannot be provided at the same time."
+            )
+
+        if auto_estimate:
+            estimate_fee = await self._signed_estimate_fee(transaction)
+            max_fee = int(estimate_fee.overall_fee * 1.1)
+
+        if max_fee is None:
+            raise ValueError("Max_fee must be specified when invoking a transaction")
+
+        return max_fee
+
+    async def _prepare_invoke_function(
+        self,
+        calls: Calls,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> InvokeFunction:
+        """
+        Takes calls and creates InvokeFunction from them
+
+        :param calls: Single call or list of calls
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        :return: InvokeFunction created from the calls (without the signature)
+        """
+        nonce = await self.get_nonce()
+
+        calldata_py = _merge_calls(ensure_iterable(calls))
+        wrapped_calldata, _ = execute_transformer_v1.from_python(*calldata_py)
+
+        transaction = InvokeFunction(
+            calldata=wrapped_calldata,
+            signature=[],
+            max_fee=0,
+            version=self._version,
+            nonce=nonce,
+            contract_address=self.address,
+        )
+
+        max_fee = await self._get_max_fee(transaction, max_fee, auto_estimate)
+
+        return dataclasses.replace(transaction, max_fee=max_fee)
+
+    async def _verify_message_hash(self, msg_hash: int, signature: List[int]) -> bool:
+        """
+        Verify a signature of a given hash
+
+        :param msg_hash: hash to be verified
+        :param signature: signature of the hash
+        :return: true if the signature is valid, false otherwise
+        """
+        calldata = [msg_hash, len(signature), *signature]
+
+        call = Call(
+            to_addr=self.address,
+            selector=get_selector_from_name("is_valid_signature"),
+            calldata=calldata,
+        )
+        try:
+            await self._client.call_contract(call=call, block_hash="pending")
+            return True
+        except ClientError as ex:
+            if re.search(r"Signature\s.+,\sis\sinvalid", ex.message):
+                return False
+            raise ex
+
+    async def _signed_estimate_fee(
```

Comment:
I'm not happy with this name :/

Reply:
It's not great, do you have an idea for an alternative?

Reply:
Not really, will think about it

Reply:
Why not just `_estimate_fee`? It is a protected method.